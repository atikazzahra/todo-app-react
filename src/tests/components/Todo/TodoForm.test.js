import { fireEvent, render, screen } from "@testing-library/react";
import TodoFrom from "#components/Todo/TodoForm.tsx";
import React from "react";

describe("TodoForm", () => {
  test("initial state form", async () => {
    const mockOnSubmitHandler = jest.fn();
    const mockOnChangeHandler = jest.fn();

    const { asFragment } = render(
      <TodoFrom
        onSubmitForm={() => {
          mockOnSubmitHandler();
        }}
        onChangeForm={() => {
          mockOnChangeHandler();
        }}
      />
    );

    expect(screen.getByTestId("todo-input")).toHaveValue("1");
    expect(screen.getByTestId("button-add")).toHaveAttribute("disabled");
    expect(asFragment()).toMatchSnapshot();
  });

  test("upate input form", async () => {
    const mockOnSubmitHandler = jest.fn();
    const mockOnChangeHandler = jest.fn();

    render(
      <TodoFrom
        onSubmitForm={() => {
          mockOnSubmitHandler();
        }}
        onChangeForm={() => {
          mockOnChangeHandler();
        }}
      />
    );

    const input = screen.getByTestId("todo-input");
    const value = "Testing";
    fireEvent.change(input, { target: { value: value } });
    expect(screen.getByTestId("todo-input")).toHaveValue(value);
    expect(screen.getByTestId("button-add")).not.toHaveAttribute("disabled");
    expect(mockOnChangeHandler).toHaveBeenCalled();
  });
});
