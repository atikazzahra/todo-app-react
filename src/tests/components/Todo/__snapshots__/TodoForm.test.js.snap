// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoForm initial state form 1`] = `
<DocumentFragment>
  <div
    class="todo-form"
  >
    <div
      class="input-group mb-3"
    >
      <input
        aria-describedby="todo-input"
        aria-label="todo-input"
        class="form-control"
        data-testid="todo-input"
        placeholder="Your new todo..."
        type="text"
        value=""
      />
      <button
        class="btn btn-secondary"
        data-testid="button-add"
        disabled=""
        type="button"
      >
        Add
      </button>
    </div>
  </div>
</DocumentFragment>
`;
